internal/service/resources/collector/collector_acm.go

```diff
@@ -535,7 +535,7 @@ func (d *ACMDataSource) convertManagedClusterToDeploymentManager(ctx context.Con
 
 	extensions, err := d.getDeploymentManagerExtensions(ctx, cluster.Name)
 	if err != nil {
-		return models.DeploymentManager{}, fmt.Errorf("failed to get extensions for cluster %s", cluster.Name)
+		return models.DeploymentManager{}, fmt.Errorf("failed to get extensions for cluster %s: %s", cluster.Name, err.Error())
```

Comment:
consider using `"%w", err` for error to preserve the error chain and allow error inspection. Unless you specifically wanted this?